src/Service.Host/client/src/components/VatCode.js

```diff
@@ -0,0 +1,194 @@
+import React, { Fragment, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from '@material-ui/core';
+import {
+    SaveBackCancelButtons,
+    InputField,
+    Loading,
+    Title,
+    ErrorCard
+} from '@linn-it/linn-form-components-library';
+import Page from '../containers/Page';
+
+function VatCode({
+    errorMessage,
+    editStatus,
+    history,
+    loading,
+    vatCodeId,
+    addVatCode,
+    setEditStatus,
+    updateVatCode,
+    ...props
+}) {
+    const [vatCode, setVatCode] = useState({});
+    const [prevVatCode, setPrevVatCode] = useState(null);
+
```

Comment:
Using hooks instead of state here. I think it looks a little tidier. prevVatCode is used in place of getDerivedStateFromProps. 

I've not created any shared hooks as using them seems to slim down the functions used in the component.